glean-core/android/src/main/java/mozilla/telemetry/glean/rust/LibGleanFFI.kt

```diff
@@ -65,10 +66,21 @@ internal fun Pointer.getRustString(): String {
 @Suppress("TooManyFunctions")
 internal interface LibGleanFFI : Library {
     companion object {
-        private val JNA_LIBRARY_NAME = "glean_ffi"
-
         internal var INSTANCE: LibGleanFFI = try {
-            val lib = Native.load(JNA_LIBRARY_NAME, LibGleanFFI::class.java) as LibGleanFFI
+            var libName = BuildConfig.JNA_LIBRARY_NAME
+            // If we're part of the application-services megazord, we may need to dynamically
+            // look up the library name in system properties, to work with custom megazords.
+            // Ref https://github.com/mozilla/application-services/blob/main/docs/design/megazords.md
+            if (libName == "megazord") {
+                val mzLibName = System.getProperty("mozilla.appservices.megazord.library")
```

Comment:
AppServices actually publishes a helper module for finding the right native library, which has [significantly more logic](https://github.com/mozilla/application-services/blob/b7839356f2fe07dee03d593e9d9d12ae91b3e062/components/support/android/src/main/java/mozilla/appservices/support/native/Helpers.kt#L160) around checking that we're loading the correct version of the compiled native code. We could take a dependency on `org.mozilla.appservices:native-support` here and use it, but that would in turn pull in a dependency on `com.google.protobuf:protobuf-javalite` which you may not want.

IMHO much of the logic in the `native-support` helper is that that useful these days because we're much better at preventing version mismatches at build time.

Reply:
Hm, the protobuf part of native-support doesn't seem to be used for the megazord handling part, so we could still strip that out if we really need to unify the loading code.

Reply:
Right. We may be able to factor it into a separate dependency on our side as well.

Reply:
> is that that

Uh, this was supposed to be "is not that"...